File: src/main/java/named/entity/recognition/RegexRecognizer.java
package named.entity.recognition;

import named.entity.NERType;

import java.util.HashMap;
import java.util.Objects;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class RegexRecognizer extends NamedEntityRecognizer {

    private final Pattern pattern;
    private final NERType nerType;

    public RegexRecognizer(final Pattern pattern, final NERType nerType){
        this.pattern = Objects.requireNonNull(pattern, "pattern cannot be null");
        this.nerType = Objects.requireNonNull(nerType, "nerType cannot be null");
    }
    @Override
    public HashMap<String, NERType> getNamedEntities(final String text) {
        Objects.requireNonNull(text);
        final Matcher matcher = pattern.matcher(text);
        final HashMap<String, NERType> namedEntities = new HashMap<>();
        while (matcher.find()){
            String namedEntity = matcher.group().toLowerCase();
            namedEntities.put(namedEntity, nerType);
        }
        return namedEntities;
    }
}

File: src/main/java/named/entity/recognition/RegexRecognizer.kt
package named.entity.recognition

import named.entity.NERType
import java.util.*
import java.util.regex.Pattern

class RegexRecognizer(
    private val pattern: Pattern,
    private val nerType: NERType
) : NamedEntityRecognizer() {

    override fun getNamedEntities(text: String): HashMap<String, NERType> {
        Objects.requireNonNull(text)
        val matcher = pattern.matcher(text)
        val namedEntities = HashMap<String, NERType>()
        while (matcher.find()) {
            val namedEntity = matcher.group().toLowerCase()
            namedEntities[namedEntity] = nerType
        }
        return namedEntities
    }
}
